### ISSUE TITLE
Big Query TIME type not handled

### ISSUE THREAD

**Issue by remydavid** [metabase/metabase#5826]
Following valid Bigquery requests : 
`
    SELECT EXTRACT(TIME
    FROM TIMESTAMP "2008-12-25 15:30:00")
    AS time
`
`
SELECT
  TIME(15, 30, 00) as time_hms
`
generates a java.lang.NullPointerException exception in Metabase 0.25.2

**Comment by salsakran** [comment]
repro'd

**Comment by camsaul** [comment]
stacktrace:

```
:error "java.lang.NullPointerException",
:stacktrace
["driver.bigquery$post_process_native$iter__48778__48782$fn__48783$iter__48804__48808$fn__48809.invoke(bigquery.clj:181)"
"query_processor.middleware.annotate_and_sort$infer_column_types$iter__27415__27419$fn__27420$fn__27421$iter__27422__27426$fn__27427.invoke(annotate_and_sort.clj:30)"
"driver$values__GT_base_type.invokeStatic(driver.clj:342)"
"driver$values__GT_base_type.invoke(driver.clj:336)"
"query_processor.middleware.annotate_and_sort$infer_column_types$iter__27415__27419$fn__27420$fn__27421.invoke(annotate_and_sort.clj:29)"
"query_processor.middleware.annotate_and_sort$infer_column_types$iter__27415__27419$fn__27420.invoke(annotate_and_sort.clj:25)"
"query_processor.middleware.annotate_and_sort$infer_column_types.invokeStatic(annotate_and_sort.clj:25)"
"query_processor.middleware.annotate_and_sort$infer_column_types.invoke(annotate_and_sort.clj:19)"
"query_processor.middleware.annotate_and_sort$annotate_and_sort$fn__27458.invoke(annotate_and_sort.clj:44)"
"query_processor.middleware.limit$limit$fn__28985.invoke(limit.clj:14)"
"query_processor.middleware.cumulative_aggregations$cumulative_aggregation$fn__28847.invoke(cumulative_aggregations.clj:46)"
"query_processor.middleware.cumulative_aggregations$cumulative_aggregation$fn__28847.invoke(cumulative_aggregations.clj:46)"
"query_processor.middleware.format_rows$format_rows$fn__28975.invoke(format_rows.clj:21)"
"query_processor.middleware.binning$update_binning_strategy$fn__27537.invoke(binning.clj:172)"
"query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__30254.invoke(results_metadata.clj:89)"
"query_processor.middleware.resolve$resolve_middleware$fn__27065.invoke(resolve.clj:359)"
"query_processor.middleware.expand$expand_middleware$fn__28741.invoke(expand.clj:550)"
"query_processor.middleware.add_row_count_and_status$add_row_count_and_status$fn__27149.invoke(add_row_count_and_status.clj:14)"
"query_processor.middleware.driver_specific$process_query_in_context$fn__28867.invoke(driver_specific.clj:12)"
"query_processor.middleware.resolve_driver$resolve_driver$fn__30265.invoke(resolve_driver.clj:14)"
"query_processor.middleware.cache$maybe_return_cached_results$fn__27618.invoke(cache.clj:146)"
"query_processor.middleware.catch_exceptions$catch_exceptions$fn__28789.invoke(catch_exceptions.clj:58)"
"query_processor$process_query.invokeStatic(query_processor.clj:126)"
"query_processor$process_query.invoke(query_processor.clj:122)"
"query_processor$run_and_save_query_BANG_.invokeStatic(query_processor.clj:234)"
"query_processor$run_and_save_query_BANG_.invoke(query_processor.clj:229)"
"query_processor$fn__30299$process_query_and_save_execution_BANG___30304$fn__30305.invoke(query_processor.clj:272)"
"query_processor$fn__30299$process_query_and_save_execution_BANG___30304.invoke(query_processor.clj:258)"
"api.dataset$fn__32183$fn__32186.invoke(dataset.clj:63)"
"api.common.internal$do_with_caught_api_exceptions.invokeStatic(internal.clj:248)"
"api.common.internal$do_with_caught_api_exceptions.invoke(internal.clj:243)"
"api.dataset$fn__32183.invokeStatic(dataset.clj:54)"
"api.dataset$fn__32183.invoke(dataset.clj:54)"
"middleware$enforce_authentication$fn__32046.invoke(middleware.clj:122)"
"api.routes$fn__43886.invokeStatic(routes.clj:61)"
"api.routes$fn__43886.invoke(routes.clj:61)"
"routes$fn__45704$fn__45705.doInvoke(routes.clj:75)"
"routes$fn__45704.invokeStatic(routes.clj:71)"
"routes$fn__45704.invoke(routes.clj:71)"
"middleware$log_api_call$fn__32145$fn__32147.invoke(middleware.clj:330)"
"middleware$log_api_call$fn__32145.invoke(middleware.clj:329)"
"middleware$add_security_headers$fn__32095.invoke(middleware.clj:245)"
"middleware$bind_current_user$fn__32050.invoke(middleware.clj:142)"
"middleware$maybe_set_site_url$fn__32099.invoke(middleware.clj:268)"]
```

**Comment by senior** [comment]
We don't currently handle time fields correctly on any database. On other databases we yield incorrect results, from this it looks like BigQuery just fails outright. The related tickets for this issue are  https://github.com/metabase/metabase/issues/5859 and https://github.com/metabase/metabase/issues/5563

**Comment by camsaul** [comment]
@senior did you ever end up merging the commits referenced here?

**Comment by senior** [comment]
I think those commits have been superseded by https://github.com/metabase/metabase/pull/6109, which adds time support for all of our drivers.

**Comment by senior** [comment]
Fixed in #6109
